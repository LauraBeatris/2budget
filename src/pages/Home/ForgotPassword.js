import React, { Component } from "react";
import { firebase } from "../../firebase/firebase";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import Helmet from "react-helmet";
import * as AuthActions from "../../actions/auth";

import { HomeContainer, FirstColumn } from "./styles-forgot";
import User from "../../assets/user.svg";

class ForgotPassword extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      success: false
    };
  }

  render() {
    const { reset_password } = this.props;

    return (
      <HomeContainer>
        <Helmet>
          <title> 2Budget | Forgot Password </title>
        </Helmet>
        <FirstColumn>
          {" "}
          <header>
            <Link id="back-button" to="/">
              &#8629; Back
            </Link>
            <h1 className="home-title">
              <span>2</span>Budget
            </h1>
          </header>
          <form
            onSubmit={ev => {
              ev.preventDefault();

              if (!this.email.value) {
                this.setState({ error: "Empty email" });
              } else {
                reset_password(this.email.value)
                  .then(() => {
                    this.setState({ error: null });

                    this.setState({ success: true });

                    setTimeout(() => {
                      this.props.history.push("/");
                    }, 1500);
                  })
                  .catch(err => {
                    console.log("Error code: ", err);
                    this.setState({ success: false });
                    // const errors = {
                    //   "auth/user-not-found": "User not found"
                    // };
                    // this.setState({ error: errors[err.code] });
                    console.log("Error message: ", err.message);
                  });
              }
            }}
            className="
            form-container"
          >
            <p id="subtitle"> Forgot your password? </p>
            <label>Type your email:</label>
            <div id="email">
              <input
                type="text"
                ref={input => (this.email = input)}
                name="email"
                placeholder="Email"
              />
              <span>
                <img src={User} alt="user-icon" />
              </span>
            </div>
            {this.state.error ? (
              <p className="error" id="error-msg">
                {this.state.error}
              </p>
            ) : null}
            {this.state.success ? (
              <p className="error" id="error-msg">
                {this.state.success}
              </p>
            ) : null}

            <div id="button-container">
              <button type="submit" className="login-button">
                Send password reset email &gt;
              </button>
            </div>
          </form>
        </FirstColumn>
      </HomeContainer>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  reset_password: email => dispatch(AuthActions.startPasswordReset(email))
});

export default connect(
  undefined,
  mapDispatchToProps
)(ForgotPassword);
