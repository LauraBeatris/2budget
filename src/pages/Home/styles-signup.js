import styled from "styled-components";

export const HomeContainer = styled.section`
  background-color: $background-color;
  padding: 8rem 18rem;

  display: flex;
  flex-direction: row;
  justify-content: space-around;

  @media screen and (max-width: 1040px) {
    margin: 4rem auto;
  }

  .google-msg {
    p span {
      margin-top: 2rem;
      font-size: 1.5rem;
    }
  }
`;

export const FirstColumn = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;
  align-items: center;

  @media screen and (max-width: 1020px) {
    text-align: center;
    width: inherit;
  }

  #email {
    margin-bottom: 2rem;
  }

  #password,
  #email {
    position: relative;

    input {
      border: 1px solid rgb(77, 81, 94);
    }
  }

  #email,
  #password {
    span {
      img {
        width: 20px;
      }

      position: absolute;
      top: 10px;
      left: 5%;

      @media screen and (max-width: 1020px) {
        left: 2.5%;
      }
    }
  }

  .home-title {
    color: rgb(77, 81, 94);
    font-size: 4.5rem;
    text-transform: uppercase;
    margin-bottom: 4rem;

    span {
      color: #6dbfcc;
      font-size: 5rem;
    }
  }

  form.form-container {
    display: flex;
    flex-direction: column;

    #subtitle {
      font-size: 2rem;
      color: rgb(77, 81, 94);
      margin-bottom: 4rem;
      font-weight: bold;
    }

    label {
      text-align: center;
      color: rgb(77, 81, 94);
      font-size: 1.5rem;
      margin-bottom: 0.5rem;
      font-weight: 700;
    }

    &--signup {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
    }

    input {
      color: #a8a9af;
      box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.2);
      font-size: 1.5rem;
      font-weight: lighter;
      padding: 1rem 2rem;
      padding-left: 4rem;
      border: none;
      border-radius: 15px;
      display: inline-block;

      &::placeholder {
        color: #a8a9af;
      }

      @media screen and (max-width: 1040px) {
        width: 75vw;
      }
    }

    #password {
      background-image: url("../../assets/lock.svg");
    }

    p#error-msg {
      font-size: 1.5rem;
      font-weight: bold;
      color: red;
      margin-top: 1rem;
    }

    p.msg-form {
      margin-top: 2rem;
      font-size: 2rem;
      color: #7f8084;
      font-weight: bold;

      cursor: pointer;

      &:hover {
        text-decoration: underline;
      }
    }

    .login-button {
      margin-top: 3rem;
      background-color: #6dbfcc;
      border: none;
      font-size: 1.8rem;
      font-weight: bold;
      padding: 2rem 4rem;
      color: white;
      border-radius: 100px;
      box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.2);
      cursor: pointer;
      text-decoration: none;
      transition: all 0.5s;

      &:hover {
        transform: translateY(-6px);
      }

      &:active {
        transform: translateY(0px);
        box-shadow: 0 2rem 2rem rgba(0, 0, 0, 0.2);
      }

      @media screen and (max-width: 1040px) {
        width: 50vw;
        margin: 3rem auto;
      }
    }
  }

  div.msg-container {
    margin-top: 5rem;
    text-align: center;

    p {
      font-size: 1.5rem;
      color: #7f8084;
      font-weight: bold;

      a {
        color: rgb(209, 77, 53);
        cursor: pointer;

        &:hover {
          text-decoration: underline;
        }
      }
    }
    .copy {
      color: rgb(209, 77, 53);
      font-weight: 700;
      margin-top: 0.5rem;

      a {
        color: inherit;
      }
    }
    @media screen and (max-width: 1040px) {
      margin: 4.5rem auto;

      p {
        font-size: 2rem;
      }
    }
  }

  input:-webkit-autofill,
  input:-webkit-autofill:hover,
  input:-webkit-autofill:focus,
  textarea:-webkit-autofill,
  textarea:-webkit-autofill:hover,
  textarea:-webkit-autofill:focus,
  select:-webkit-autofill,
  select:-webkit-autofill:hover,
  select:-webkit-autofill:focus {
    -webkit-box-shadow: 0 0 0px 1000px #fff inset;
    transition: background-color 5000s ease-in-out 0s;
  }

  #button-container {
    display: flex;
    justify-content: center;
    align-items: center;

    button {
      width: 70%;
    }
  }

  #form-container-signup {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    input {
      &:first-child {
        margin-bottom: 4rem;
      }

      color: #a8a9af;
      box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.2);
      font-size: 1.5rem;
      font-weight: lighter;
      padding: 1rem 2rem;
      padding-left: 4rem;
      border: none;
      border-radius: 15px;
      display: inline-block;

      &::placeholder {
        color: #a8a9af;
      }

      @media screen and (max-width: 1040px) {
        width: 75vw;
      }
    }
  }
`;

export const SecondColumn = styled.div`
  img {
    border-radius: 100px;
    margin-left: 100px;
    height: 50rem;
  }

  @media screen and (max-width: 1040px) {
    display: none;
  }
`;

export const Google = styled.a`
  font-size: 1.5rem;
  color: #7f8084;
  margin-top: 2rem;
  font-weight: 700;

  span {
    text-decoration: underline;
    color: rgb(209, 77, 53);
    cursor: pointer;
  }
`;
