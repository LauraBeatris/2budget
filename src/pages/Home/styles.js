import styled from "styled-components";

export const HomeContainer = styled.section`
  background-color: $background-color;
  padding: 8rem 18rem;

  display: flex;
  flex-direction: row;
  justify-content: space-around;

  @media screen and (max-width: 1040px) {
    margin: 4rem auto;
  }

  .google-msg {
    p span {
      margin-top: 2rem;
      font-size: 1.5rem;
    }
  }
`;

export const FirstColumn = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;
  width: 25%;

  @media screen and (max-width: 1020px) {
    text-align: center;
    width: inherit;
  }

  #password,
  #email {
    position: relative;
    margin: 0 auto;

    input {
      border: 1px solid rgb(77, 81, 94);
    }
  }

  #email,
  #password {
    span {
      img {
        width: 20px;
      }

      position: absolute;
      top: 10px;
      left: 5%;

      @media screen and (max-width: 1020px) {
        left: 2.5%;
      }
    }
  }

  .home-title {
    color: rgb(77, 81, 94);
    font-size: 4.5rem;
    text-transform: uppercase;
    margin-bottom: 4rem;

    span {
      color: #6dbfcc;
      font-size: 5rem;
    }
  }

  form.form-container {
    display: flex;
    flex-direction: column;

    &--signup {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
    }

    input {
      &:first-child {
        margin-bottom: 4rem;
      }

      color: #a8a9af;
      box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.2);
      font-size: 1.5rem;
      font-weight: lighter;
      padding: 1rem 2rem;
      padding-left: 4rem;
      border: none;
      border-radius: 15px;
      display: inline-block;

      &::placeholder {
        color: #a8a9af;
      }

      @media screen and (max-width: 1040px) {
        width: 75vw;
      }
    }

    #password {
      background-image: url("../../assets/lock.svg");
    }

    p#error-msg {
      font-size: 1.5rem;
      font-weight: bold;
      color: red;
      margin: 1rem 0rem;
    }

    p.msg-form {
      margin-top: 0.5rem;
      font-size: 2rem;
      color: #7f8084;
      font-weight: bold;

      cursor: pointer;

      &:hover {
        text-decoration: underline;
      }
    }

    .login-button {
      margin-top: 2rem;
      background-color: #6dbfcc;
      border: none;
      font-size: 1.8rem;
      font-weight: bold;
      padding: 2rem 4rem;
      color: white;
      border-radius: 100px;
      box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.2);
      cursor: pointer;
      text-decoration: none;
      transition: all 0.5s;

      &:hover {
        transform: translateY(-6px);
      }

      &:active {
        transform: translateY(0px);
        box-shadow: 0 2rem 2rem rgba(0, 0, 0, 0.2);
      }

      @media screen and (max-width: 1040px) {
        width: 50vw;
        margin: 1.5rem auto;
      }
    }
  }

  div.msg-container {
    margin-top: 5rem;
    text-align: center;

    p {
      font-size: 1.5rem;
      color: #7f8084;
      font-weight: bold;

      a {
        color: rgb(209, 77, 53);
        cursor: pointer;

        &:hover {
          text-decoration: underline;
        }
      }
    }
    .copy {
      color: rgb(209, 77, 53);
      font-weight: 700;
      margin-top: 0.5rem;

      a {
        color: inherit;
      }
    }
    @media screen and (max-width: 1040px) {
      margin: 10rem auto;

      p {
        font-size: 1.9rem;
      }
    }
  }

  input:-webkit-autofill,
  input:-webkit-autofill:hover,
  input:-webkit-autofill:focus,
  textarea:-webkit-autofill,
  textarea:-webkit-autofill:hover,
  textarea:-webkit-autofill:focus,
  select:-webkit-autofill,
  select:-webkit-autofill:hover,
  select:-webkit-autofill:focus {
    -webkit-box-shadow: 0 0 0px 1000px #fff inset;
    transition: background-color 5000s ease-in-out 0s;
  }

  #button-container {
    display: flex;
    justify-content: center;
    align-items: center;

    button {
      width: 70%;
    }
  }

  #login-msg {
    margin: 1rem 0;
    font-size: 2rem;
    color: #7f8084;
    font-weight: bold;
    font-size: 1.6rem;
  }

  #social-medias-container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  #social-medias {
    display: flex;
    flex-direction: row;
    justify-content: center;

    img {
      width: 2.5rem;
      &:not(:first-child) {
        margin-left: 1rem;
      }

      cursor: pointer;
      transition: all 0.25s;

      &:hover,
      &:focus {
        transform: scale(1.1);
      }
    }
  }

  #form-container-signup {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    input {
      &:first-child {
        margin-bottom: 4rem;
      }

      color: #a8a9af;
      box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.2);
      font-size: 1.5rem;
      font-weight: lighter;
      padding: 1rem 2rem;
      padding-left: 4rem;
      border: none;
      border-radius: 15px;
      display: inline-block;

      &::placeholder {
        color: #a8a9af;
      }

      @media screen and (max-width: 1040px) {
        width: 75vw;
      }
    }
  }
`;

export const SecondColumn = styled.div`
  img {
    border-radius: 100px;
    margin-left: 100px;
    height: 50rem;
  }

  @media screen and (max-width: 1040px) {
    display: none;
  }
`;

export const Google = styled.a`
  font-size: 1.5rem;
  color: #7f8084;
  margin-top: 2rem;
  font-weight: 700;

  span {
    text-decoration: underline;
    color: rgb(209, 77, 53);
    cursor: pointer;
  }
`;
