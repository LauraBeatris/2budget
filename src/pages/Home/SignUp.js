import React, { Component } from "react";
import { HomeContainer, FirstColumn } from "./styles-signup";
import User from "../../assets/user.svg";
import Lock from "../../assets/lock.svg";

import { firebase } from "../../firebase/firebase";
import { connect } from "react-redux";
import * as AuthActions from "../../actions/auth";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null
    };
  }

  render() {
    const { google_auth } = this.props;
    return (
      <HomeContainer>
        <FirstColumn>
          {" "}
          <header>
            <h1 className="home-title">
              <span>2</span>Budget
            </h1>
          </header>
          <form
            onSubmit={ev => {
              ev.preventDefault();

              if (!this.password.value) {
                this.setState({ error: "Empty password" });
              } else if (!this.email.value) {
                this.setState({ error: "Empty email" });
              } else {
                firebase
                  .auth()
                  .signInWithEmailAndPassword(
                    this.email.value,
                    this.password.value
                  )
                  .then(res => {
                    const token = res.credential.accessToken;
                    const user = res.user;

                    this.setState({ error: null });
                    this.props.history.push("/dashboard");
                  })
                  .catch(err => {
                    console.log("Error code: ", err.code);

                    const errors = {
                      "auth/user-not-found": "User not found"
                    };
                    this.setState({ error: errors[err.code] });
                    console.log("Error message: ", err.message);
                  });
              }
            }}
            className="
            form-container"
          >
            <p id="subtitle"> Create your account </p>
            <label>Type your email:</label>
            <div id="email">
              <input
                type="text"
                ref={input => (this.email = input)}
                name="email"
                placeholder="Email"
              />
              <span>
                <img src={User} alt="user-icon" />
              </span>
            </div>
            <label>Type your password:</label>
            <div id="password">
              <input
                id="password"
                type="password"
                ref={input => (this.password = input)}
                name="password"
                placeholder="Password"
              />
              <span>
                <img id="user" src={Lock} alt="user-cion" />
              </span>
            </div>
            <p className="error" id="error-msg">
              {" "}
              {this.state.error}{" "}
            </p>

            <div id="button-container">
              <button type="submit" className="login-button">
                Sign Up &gt;
              </button>
            </div>
          </form>
          <div className="msg-container">
            <p className="copy">
              Copyright Â©{" "}
              <a
                href="https://github.com/LauraBeatris"
                target="_blank"
                rel="noopener noreferrer"
              >
                Laura Beatris{" "}
              </a>
            </p>
          </div>
        </FirstColumn>
      </HomeContainer>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  google_auth: () => dispatch(AuthActions.startLoginGoogle())
});

export default connect(
  undefined,
  mapDispatchToProps
)(Home);
