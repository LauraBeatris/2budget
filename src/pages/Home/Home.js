import React, { Component } from "react";
import { HomeContainer, FirstColumn, SecondColumn, Google } from "./styles";
import BudgetIllustration from "../../assets/budget_illustration.png";
import User from "../../assets/user.svg";
import Lock from "../../assets/lock.svg";

import { firebase } from "../../firebase/firebase";
import { connect } from "react-redux";
import * as AuthActions from "../../actions/auth";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null
    };
  }

  render() {
    const { google_auth } = this.props;
    return (
      <HomeContainer>
        <FirstColumn>
          {" "}
          <header>
            <h1 className="home-title">
              <span>2</span>Budget
            </h1>
          </header>
          <form
            onSubmit={ev => {
              ev.preventDefault();

              if (!this.password.value) {
                this.setState({ error: "Empty password" });
              } else if (!this.email.value) {
                this.setState({ error: "Empty email" });
              } else {
                firebase
                  .auth()
                  .signInWithEmailAndPassword(
                    this.email.value,
                    this.password.value
                  )
                  .then(res => {
                    const token = res.credential.accessToken;
                    const user = res.user;

                    this.setState({ error: null });
                    this.props.history.push("/dashboard");
                  })
                  .catch(err => {
                    console.log("Error code: ", err.code);

                    const errors = {
                      "auth/user-not-found": "User not found"
                    };
                    this.setState({ error: errors[err.code] });
                    console.log("Error message: ", err.message);
                  });
              }
            }}
            className="form-container"
          >
            <div id="email">
              <input
                type="text"
                ref={input => (this.email = input)}
                name="email"
                placeholder="Email"
              />
              <span>
                <img src={User} alt="user-cion" />
              </span>
            </div>
            <div id="password">
              <input
                id="password"
                type="password"
                ref={input => (this.password = input)}
                name="password"
                placeholder="Password"
              />
              <span>
                <img id="user" src={Lock} alt="user-cion" />
              </span>
            </div>
            <p className="error" id="error-msg">
              {" "}
              {this.state.error}{" "}
            </p>
            <p className="msg-form">Forgot Password?</p>
            <button type="submit" className="login-button">
              Login &gt;
            </button>
          </form>
          <Google
            onClick={ev => {
              google_auth()
                .then(res => {
                  // This gives you a Google Access Token. You can use it to access the Google API.
                  var token = res.credential.accessToken;
                  // The signed-in user info.
                  var user = res.user;

                  // Storing user data
                  localStorage.setItem({ "auth-token": token });
                  localStorage.getItem({ user });

                  this.setState({ error: null });
                  // Redirecting the user
                  this.props.history.push("/dashboard");
                })
                .catch(err => {
                  var errorCode = err.code;
                  var errorMessage = err.message;
                  var email = err.email;
                  var credential = err.credential;

                  const errors = {
                    "auth/user-not-found": "User not found"
                  };

                  //console.log(errors[err.code]);

                  this.setState({ error: errors[errorCode] });

                  console.log(err.code);
                  console.log("User email used", email);
                  console.log("Credential", credential);
                });
            }}
            className="msg-google"
          >
            Or <span>Login with google</span>
          </Google>
          <div className="msg-container">
            <p>
              Don't have an account? <span>Sign up</span>
            </p>
            <p className="copy">
              Copyright Â©{" "}
              <a
                href="https://github.com/LauraBeatris"
                target="_blank"
                rel="noopener noreferrer"
              >
                Laura Beatris{" "}
              </a>
            </p>
          </div>
        </FirstColumn>

        <SecondColumn>
          <img src={BudgetIllustration} alt="budget-illustration" />
        </SecondColumn>
      </HomeContainer>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  google_auth: dispatch(AuthActions.startLoginGoogle)
});

export default connect(
  undefined,
  mapDispatchToProps
)(Home);
