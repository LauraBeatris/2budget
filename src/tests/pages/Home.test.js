import React from "react";
import Home from "../../pages/Home/Home";
import { shallow } from "enzyme";

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Home />);
});

test("should render homepage correctly", () => {
  const wrapper = shallow(<Home />);
  expect(wrapper).toMatchSnapshot();
});
