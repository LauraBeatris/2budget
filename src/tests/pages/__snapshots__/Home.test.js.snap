// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render homepage correctly 1`] = `
<.sc-bdVaJa>
  <.sc-bwzfXH>
     
    <header>
      <h1
        className="home-title"
      >
        <span>
          2
        </span>
        Budget
      </h1>
    </header>
    <form
      className="form-container"
    >
      <div
        id="email"
      >
        <input
          name="email"
          placeholder="Email"
          type="text"
        />
        <span>
          <img
            alt="user-cion"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </div>
      <div
        id="password"
      >
        <input
          id="password"
          name="password"
          placeholder="Password"
          type="password"
        />
        <span>
          <img
            alt="user-cion"
            id="user"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </div>
      <p
        className="msg-form"
      >
        Forgot Password?
      </p>
      <Link
        className="login-button"
        to="/dashboard"
      >
        Login &gt;
      </Link>
    </form>
    <div
      className="msg-container"
    >
      <p>
        Don't have an account? 
        <span>
          Sign up
        </span>
      </p>
      <p
        className="copy"
      >
        Copyright Â©
         
        <a
          href="https://github.com/LauraBeatris"
          rel="noopener noreferrer"
          target="_blank"
        >
          Laura Beatris
           
        </a>
      </p>
    </div>
  </.sc-bwzfXH>
  <.sc-htpNat>
    <img
      alt="budget-illustration"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </.sc-htpNat>
</.sc-bdVaJa>
`;
